cmd/peer tests fail due to port conflict

sykesm opened the pull request:
The PluginLoadingFailure test starts the peer without overriding the chaincode listen address. This results in a port conflict when running tests in parallel.

This change extends the pattern in the test that attempts to reserve an ephemeral port and then configures the peer to use it.

tuand27613 commented:
Signed-off-by: Tuan Dang <tdang@us.ibm.com>

tuand27613 commented:
unit tests all pass.

behave tests pass except for issue #739. But the state transfer tests pass when I un them separately.

sykesm committed:
cmd/peer tests fail due to port conflict

The PluginLoadingFailure test starts the peer without overriding the
chaincode listen address. This results in a port conflict when running
tests in parallel.

This change extends the pattern in the test that attempts to reserve an
ephemeral port and then configures the peer to use it.

Signed-off-by: Matthew Sykes <matthew.sykes@gmail.com>

sykesm commented:
Opened [FAB-17535](https://jira.hyperledger.org/browse/FAB-17535) for the integration test flake.
